library/include/G4CMPGeometryUtils.hh

```diff
@@ -60,72 +62,69 @@ namespace G4CMP {
   
   void RotateToGlobalPosition(const G4VTouchable* touch,
 			      G4ThreeVector& pos);
-
-  G4ThreeVector GetGeneralizedSurfaceNormal(const G4ThreeVector& surfNorm,
-					    const G4ThreeVector& incMomDir );
-
-  G4ThreeVector GetSurfaceNormal(const G4Step& step);
+					    
+  G4ThreeVector GetSurfaceNormal(const G4Step& step,
+				 const G4ThreeVector& incMomDir = nullVec );
 
   G4double Get2DSafety(const G4VTouchable* motherTouch,
-		       G4ThreeVector pos,
-		       G4ThreeVector momDir,
+		       const G4ThreeVector& pos_in,
+		       const G4ThreeVector& momDir_in,
 		       bool safetyFromABoundary,
 		       bool forceSweepSafetyForDaughters = false,
-		       G4ThreeVector surfaceNorm = G4ThreeVector(0,0,0),
-		       G4ThreeVector tangVect1 = G4ThreeVector(0,0,0),
-		       G4ThreeVector tangVect2 = G4ThreeVector(0,0,0));
-
+		       const G4ThreeVector& surfaceNorm_in = nullVec,
```

Comment:
What does the "_in" suffix mean?  I thought we had agreed the surface normal was always (the conversion used by G4VSolid and G4Navigator) outward.

Reply:
Oh, I see.  You're using "_in" to represent the input arguments, as opposed to internal copies.  I see.  I need to look at these use cases a bit more to understand.